v2/internal/reconcilers/azuresql/connector.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package sql
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	ctrlconversion "sigs.k8s.io/controller-runtime/pkg/conversion"
+
+	asosql "github.com/Azure/azure-service-operator/v2/api/sql/v1"
+	sql "github.com/Azure/azure-service-operator/v2/api/sql/v1api20211101/storage"
+	"github.com/Azure/azure-service-operator/v2/internal/resolver"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime/conditions"
+)
+
+type Connector interface {
+	CreateOrUpdate(ctx context.Context) error
+	Delete(ctx context.Context) error
+	Exists(ctx context.Context) (bool, error)
+}
+
+type ownerDetails struct {
+	fqdn     string
+	database string
+}
+
+func getOwnerDetails(ctx context.Context, resourceResolver *resolver.Resolver, user *asosql.User) (ownerDetails, error) {
+	// Get the owner - at this point it must exist
+	owner, err := resourceResolver.ResolveOwner(ctx, user)
+	if err != nil {
+		return ownerDetails{}, err
+	}
+
+	hierarchy, err := resourceResolver.ResolveResourceHierarchy(ctx, owner.Owner)
+	if err != nil {
+		return ownerDetails{}, err
+	}
+
+	if len(hierarchy) != 3 {
+		return ownerDetails{}, errors.Errorf("failed to look up ownerDetails: expected resource hierarchy len=3 but was %d", len(hierarchy))
+	}
+
+	genericServer := hierarchy[1]
+	genericDatabase := hierarchy[2]
+
+	server, ok := genericServer.(*sql.Server)
+	if !ok {
+		return ownerDetails{}, errors.Errorf("owner's owner was not type Server, instead: %T", genericServer)
+	}
+
+	database, ok := genericDatabase.(*sql.ServersDatabase)
+	if !ok {
+		return ownerDetails{}, errors.Errorf("owner was not type ServersDatabase, instead: %T", genericDatabase)
+	}
+
+	// Magical assertion to ensure that this is still the storage type
+	var _ ctrlconversion.Hub = &sql.ServersDatabase{}
+	// Magical assertion to ensure that this is still the storage type
+	var _ ctrlconversion.Hub = &sql.Server{}
+
+	// TODO: Possibly we want to source this from a configmap rather than reading it directly off of our parent?
```

Comment:
Remove this TODO - either use a configmap or don't...

Reply:
To be consistent with the other SQL users, have decided not to do this for now.

We can support this in a nonbreaking way in the future if we want to.